Build failure: man-pages (Darwin)

### Steps To Reproduce

Steps to reproduce the behavior:
1. `nix build "github:NixOS/nixpkgs/master#man-pages"`

Reproduced on Nixpkgs commit 68ead29211bb9890f494af3a8b5643a831d9d4c4

### Build log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/a4rsd0p48z6qggksskfxzskcqy4hk6lg-man-pages-6.7.tar.xz
source root is man-pages-6.7
setting SOURCE_DATE_EPOCH to timestamp 1710871633 of file man-pages-6.7/share/mk/src.mk
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "installPhase" }
Running phase: installPhase
install flags: -j8 SHELL=/nix/store/9wgn14v41bks2yb74jbmiiyi7y13334w-bash-5.2p26/bin/bash SHELL=env prefix=/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7 install
bash: line 1: locale: command not found
MKDIR		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man1/
MKDIR		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man2/
MKDIR		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/
MKDIR		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man4/
MKDIR		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man5/
MKDIR		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man6/
MKDIR		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man7/
MKDIR		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man8/
MKDIR		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man2type/
INSTALL		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/a64l.3
INSTALL		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/abort.3
INSTALL		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/abs.3
INSTALL		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/acos.3
INSTALL		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/acosf.3
INSTALL		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/acosh.3
INSTALL		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/acoshf.3
INSTALL		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/acoshl.3
install: skipping file '/dev/stdin', as it was replaced while being copied
INSTALL		/nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/acosl.3
install: skipping file '/dev/stdin', as it was replaced while being copied
install: skipping file '/dev/stdin', as it was replaced while being copied
make: *** [share/mk/install/man.mk:54: /nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/abort.3] Error 1
make: *** Waiting for unfinished jobs....
install: skipping file '/dev/stdin', as it was replaced while being copied
install: skipping file '/dev/stdin', as it was replaced while being copied
make: *** [share/mk/install/man.mk:54: /nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/a64l.3] Error 1
make: *** [share/mk/install/man.mk:54: /nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/acos.3] Error 1
make: *** [share/mk/install/man.mk:54: /nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/acosf.3] Error 1
make: *** [share/mk/install/man.mk:54: /nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/abs.3] Error 1
install: skipping file '/dev/stdin', as it was replaced while being copied
install: skipping file '/dev/stdin', as it was replaced while being copied
make: *** [share/mk/install/man.mk:54: /nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/acosh.3] Error 1
make: *** [share/mk/install/man.mk:54: /nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/acoshf.3] Error 1
install: skipping file '/dev/stdin', as it was replaced while being copied
make: *** [share/mk/install/man.mk:54: /nix/store/61x5w8m1vd75rzs02w8agjr2mf3n0vlf-man-pages-6.7/share/man/man3/acoshl.3] Error 1
```

### Notify maintainers

@trofi 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.4.0, macOS 14.4.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Lix, like Nix) 2.90.0pre20240518_139d31f
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
output here
```

(Note: it says "Lix" as that's the primary Nix on my system, but I also reproduced this with Nix 2.18)

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/data/documentation/man-pages/default.nix
+++ b/pkgs/data/documentation/man-pages/default.nix
@@ -2,16 +2,14 @@
 
 stdenv.mkDerivation rec {
   pname = "man-pages";
-  version = "6.7";
+  version = "6.8";
 
   src = fetchurl {
     url = "mirror://kernel/linux/docs/man-pages/${pname}-${version}.tar.xz";
-    hash = "sha256-gkA61LwXqtuST2hji3nWkwssvVUVMSSKepaId52077I=";
+    hash = "sha256-ucawpCD4ORSL4EsvwTqFaSMTco1U1HxpyKE4N5Zl0iY=";
   };
 
   makeFlags = [
-    # Clobber /usr/bin/env with the one in PATH.
-    "SHELL=env"
     "prefix=${placeholder "out"}"
   ];
 

